src/sage/rings/semirings/tropical_mpolynomial.py

```diff
@@ -0,0 +1,605 @@
+            Polynomial Semiring in x, y over Rational Field
+        """
+        from sage.rings.polynomial.multi_polynomial import MPolynomial
+        if isinstance(x, TropicalMPolynomial):
+            if x.parent() is not self:
+                raise ValueError(f"can not convert {x} to {self}")
+        if isinstance(x, MPolynomial):
+            if x.parent().variable_names() == self.variable_names():
+                x = x.dict()
+            else:
+                raise ValueError(f"can not convert {x} to {self}")
+        elif (x in self.base().base_ring()) or (x in self.base()):
+            term = [0]*self.ngens()
+            x = {tuple(term): x}
+        if isinstance(x, dict):
+            for key, value in x.items():  # convert coefficient to tropical
+                x[key] = self.base()(value)
+        return self.element_class(self, x)
+
+    @cached_method
+    def one(self):
+        r"""
+        Return the multiplicative identity of ``self``.
+
+        EXAMPLES::
+
+            sage: T = TropicalSemiring(QQ)
+            sage: R = PolynomialRing(T, 'x')
+            sage: R.one()
+            0
+        """
+        return self._element_constructor_(self.base().one())
+
+    @cached_method
+    def zero(self):
+        r"""
+        Return the additive identity of ``self``.
+
+        EXAMPLES::
+
+            sage: T = TropicalSemiring(QQ)
+            sage: R = PolynomialRing(T, 'x')
+            sage: R.zero()
+            +infinity
+        """
+        return self._element_constructor_(self.base().zero())
+
+    def _repr_(self):
+        r"""
+        Return a string representation of ``self``.
+
+        EXAMPLES::
+
+            sage: T = TropicalSemiring(RR)
+            sage: R.<u,v,w> = PolynomialRing(T); R
+            Multivariate Tropical Polynomial Semiring in u, v, w over Real
+            Field with 53 bits of precision
+        """
+        if self._ngens == 0:
+            return (f"Multivariate Tropical Polynomial Semiring in no variables"
+                    f" over {self.base_ring().base_ring()}")
+        return (f"Multivariate Tropical Polynomial Semiring in {', '.join(self.variable_names())}"
+                f" over {self.base_ring().base_ring()}")
+
+    def random_element(self, degree=2, terms=None, choose_degree=False,
+                       *args, **kwargs):
+        r"""
+        Return a random multivariate tropical polynomial from ``self``.
+
+        OUTPUT: a :class:`TropicalMPolynomial`
+
+        .. SEEALSO::
+
+            :meth:`sage.rings.polynomial.multi_polynomial_ring_base.MPolynomialRing_base.random_element`
+
+        EXAMPLES::
+
+            sage: T = TropicalSemiring(QQ)
+            sage: R.<a,b,c> = PolynomialRing(T)
+            sage: f = R.random_element()
+            sage: f.degree() <= 2
+            True
+            sage: len(list(f)) <= 5
+            True
+            sage: f.parent() is R
+            True
```

Comment:
This should also test some of the options (that make sense for tropical polynomials).